modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/LeftSidebar/LeftSidebarTreeView.tsx

```diff
@@ -0,0 +1,394 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import {Text, TreeView} from '@clayui/core';
+import {ClayDropDownWithItems} from '@clayui/drop-down';
+import Icon from '@clayui/icon';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import {
+	API,
+	getLocalizableLabel,
+	stringIncludesQuery,
+} from '@liferay/object-js-components-web';
+import classNames from 'classnames';
+import {openToast, sub} from 'frontend-js-web';
+import React, {useMemo} from 'react';
+import {Node, useStore, useZoomPanHelper} from 'react-flow-renderer';
+
+import './LeftSidebar.scss';
+import {useObjectFolderContext} from '../ModelBuilderContext/objectFolderContext';
+import {TYPES} from '../ModelBuilderContext/typesEnum';
+import {LeftSidebarItem, LeftSidebarObjectDefinitionItem} from '../types';
+
+const TYPES_TO_SYMBOLS = {
+	linkedObjectDefinition: 'link',
+	objectDefinition: 'catalog',
+	objectFolder: 'folder',
+};
+
+export default function LeftSidebarTreeView({
+	query,
+	showActions,
+}: {
+	query: string;
+	showActions?: boolean;
+}) {
+	const [
+		{leftSidebarItems, selectedObjectFolder},
+		dispatch,
+	] = useObjectFolderContext();
+	const store = useStore();
+	const {setCenter} = useZoomPanHelper();
+
+	const changeObjectDefinitionNodeViewButton = (
+		hiddenObjectDefinitionNode: boolean,
+		dispatch: Function
+	) => (
+		<ClayButtonWithIcon
+			aria-label={
+				hiddenObjectDefinitionNode
+					? Liferay.Language.get('hidden')
+					: Liferay.Language.get('show')
+			}
+			displayType="unstyled"
+			onClick={(event) => {
+				event.stopPropagation();
+				dispatch();
+			}}
+			symbol={hiddenObjectDefinitionNode ? 'hidden' : 'view'}
+		/>
+	);
+
+	const filteredLeftSidebarItems = useMemo(() => {
+		return leftSidebarItems.map((sidebarItem) => {
+			if (!sidebarItem.leftSidebarObjectDefinitionItems) {
+				return sidebarItem;
+			}
+
+			const newObjectDefinitions = sidebarItem.leftSidebarObjectDefinitionItems.filter(
+				(leftSidebarObjectDefinitionItem) =>
```

Comment:
The return of this filter method will be a `leftSidebarObjectDefinitionItems` so we can rename the `newObjectDefinitions`.